src/pudl/analysis/allocate_net_gen.py

```diff
@@ -731,6 +1028,121 @@ def allocate_net_gen_by_gen_esc(gen_pm_fuel):
+        / x.fuel_consumed_mmbtu_bf_tbl_pm_fuel,
+        frac=lambda x: x.frac_fuel,
+    )
+    # _ = _test_frac(all_gen)
+
+    some_gen = some_gen.assign(
+        # fraction of the fuel consumption that should go to the generators that
+        # report in the boiler fuel table
+        frac_from_bf_tbl=lambda x: x.fuel_consumed_mmbtu_bf_tbl_pm_fuel
+        / x.fuel_consumed_mmbtu_gf_tbl,
+        # for records within these mix groups that do have fuel consumption in the
+        # bf table..
+        frac_fuel=lambda x: x.fuel_consumed_mmbtu_bf_tbl
+        / x.fuel_consumed_mmbtu_bf_tbl_pm_fuel,  # generator based fuel from bf table
+        frac_bf=lambda x: x.frac_fuel * x.frac_from_bf_tbl,
+        # fraction of fuel that does not show up in the bf table
+        # set minimum fraction to zero so we don't get negative fuel
+        frac_missing_from_bf_tbl=lambda x: np.where(
+            (x.frac_from_bf_tbl < 1), (1 - x.frac_from_bf_tbl), 0
+        ),
+        capacity_mw_missing_from_bf_tbl=lambda x: np.where(
+            x.in_bf_tbl, 0, x.capacity_mw
+        ),
+        frac_cap=lambda x: x.frac_missing_from_bf_tbl
+        * (x.capacity_mw_missing_from_bf_tbl / x.capacity_mw_fuel_in_bf_tbl_group),
+        # the real deal
+        # this could aslo be `x.frac_bf + x.frac_cap` because the frac_bf
+        # should be 0 for any generator that does not have fuel in the bf_tbl
+        # and frac_cap should be 0 for any generator that has fuel in the
+        # bf_tbl.
+        frac=lambda x: np.where(x.in_bf_tbl, x.frac_bf, x.frac_cap),
+    )
+    # _ = _test_frac(some_gen)
+
+    # Calculate what fraction of the total capacity is associated with each of
+    # the generators in the grouping.
+    bf_only = bf_only.assign(
+        frac_cap=lambda x: x.capacity_mw / x.capacity_mw_unit_fuel,
+        frac=lambda x: x.frac_cap,
+    )
+    # _ = _test_frac(bf_only)
+
+    # squish all of these methods back together.
+    fuel_alloc = pd.concat([all_gen, some_gen, bf_only])
+    # null out the inf's
+    fuel_alloc.loc[abs(fuel_alloc.frac) == np.inf] = np.NaN
+    # _ = _test_frac(fuel_alloc)
+
+    # replace the placeholder 0.001 values with zero before allocating
+    # since some of these may have been aggregated, well, flag any values less than 0.01
+    fuel_alloc.loc[
+        fuel_alloc["fuel_consumed_mmbtu_gf_tbl"] < 0.01, "fuel_consumed_mmbtu_gf_tbl"
+    ] = 0
+    fuel_alloc.loc[
+        fuel_alloc["fuel_consumed_for_electricity_mmbtu_gf_tbl"] < 0.01,
+        "fuel_consumed_for_electricity_mmbtu_gf_tbl",
+    ] = 0
+
+    # do the allocating-ing!
+    fuel_alloc = (
+        fuel_alloc.assign(
+            # we could x.fuel_consumed_mmbtu_bf_tbl.fillna here if we wanted to
+            # take the net gen
+            fuel_consumed_mmbtu=lambda x: x.fuel_consumed_mmbtu_gf_tbl * x.frac,
+            fuel_consumed_for_electricity_mmbtu=lambda x: x.fuel_consumed_for_electricity_mmbtu_gf_tbl
```

Comment:
I'm vv glad you added `fuel_consumed_for_electricity_mmbtu`!

Do you think there is any difference between these two fuel columns that should lead to us treating them differently? (for the record I very much hope the answer to this question is "NO IT SHOULD BE THE SAME" for simplicity)